smac/intensification/intensification.py

```diff
@@ -588,6 +699,202 @@ def get_next_challenger(self,
         # return currently running challenger
         return self.current_challenger, False
 
+    def get_next_run(self,
+                     challengers: typing.Optional[typing.List[Configuration]],
+                     incumbent: Configuration,
+                     chooser: typing.Optional[EPMChooser],
+                     run_history: RunHistory,
+                     repeat_configs: bool = True,
+                     ) -> RunInfo:
+        """
+        Wrapper around query challengers, to get the next configuration
+        to run, as well as updating the internal state of the intensifier.
+
+        If query challenger suggest a new iteration through a None challenger,
+        this function calls itself recursively until a valid configuration is
```

Comment:
Would it be possible to make this function non-iterative? It's already quite complicated and I have the impression that the recursiveness adds another layer of complexity on top.

Reply:
Yes, I am gonna push a code that will clearly align more clearly to the intensification procedure. 

Please let me know if it required further clarification.